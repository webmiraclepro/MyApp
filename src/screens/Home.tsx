import React, { Component, Fragment } from 'react'
import LinearGradient from 'react-native-linear-gradient'
import {
  StatusBar,
  ImageBackground,
  StyleSheet,
} from 'react-native'
import {
  NavigationScreenConfig,
  NavigationScreenOptions,
} from 'react-navigation'
import R from 'res'
import { PlayButton } from '../components'

class HomeScreen extends Component {
  static navigationOptions: NavigationScreenConfig<NavigationScreenOptions> = ({ navigation }) => ({
    title: navigation.getParam('title', 'My Awesome App'),
    headerTintColor: '#fff',
    headerTitleStyle: {
      fontWeight: 'bold',
    },
    headerBackground: (
      <LinearGradient
        start={{ x: 0, y: 1 }}
        end={{ x: 1, y: 0 }}
        colors={['#6f5ce4', '#1885ef', '#8ac5f6']}
        style={StyleSheet.absoluteFill}
      >
      </LinearGradient>
    )
  })

  render() {
    return (
      <Fragment>
        <StatusBar barStyle='dark-content' backgroundColor='white' />
        <ImageBackground source={R.images.background} style={StyleSheet.absoluteFill}>
          <PlayButton />
        </ImageBackground>
      </Fragment>
    )
  }
}

export default HomeScreen
