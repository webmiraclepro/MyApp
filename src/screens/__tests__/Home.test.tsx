import 'react-native'
import React from 'react'
import { mount, shallow } from 'enzyme'

import Home from '../Home'

describe('<Home />', () => {
  const PARAMS: { [key: string]: string } = {
    title: 'CUSTOM_TITLE',
  }

  const defaultProps = {
    navigation: {
      getParam: (param: string) => {
        return PARAMS[param]
      },
    },
  } as any

  it('should render', () => {
    const rendered = mount(<Home {...defaultProps} />)
    expect(rendered).toBeTruthy()
  })

  it('renders correctly', () => {
    const wrapper = shallow(<Home {...defaultProps} />)
    expect(wrapper).toMatchSnapshot()
  })
})
