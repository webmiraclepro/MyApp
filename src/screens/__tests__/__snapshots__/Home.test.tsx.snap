// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    backgroundColor="white"
    barStyle="dark-content"
    showHideTransition="fade"
  />
  <ImageBackground
    source={
      Object {
        "testUri": "../../../src/res/images/background.jpg",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <PlayButton />
  </ImageBackground>
</Fragment>
`;
